CondeNast/launch-vehicle-fbm
# Handle a wider variety of messages more gracefully

## Why are we doing this?

Facebook uses webhooks not only for Messenger, but for all sorts of events. Currently, if a request comes into the bot and it's not a Messenger event, it'll throw an error. Instead of throwing an error, it should just ignore non Messenger events.

* Handles non-Messenger Facebook messages that can come in
* some YAGNI deletions

closes #73 

* [x] DRY the tests 

## Did you document your work?

none necessary, internal changes

## How can someone test these changes?

Steps to manually verify the change:

1. `npm install`
1. `npm test`

You can also

1. `npm link` this to an existing bot
2. go the the developer site (example: https://developers.facebook.com/apps/1156126794463877/webhooks/ )
2.  hit the "Test" button
2. your bot should not throw a 500

## What possible risks or adverse effects are there?

* none

## What are the follow-up tasks?

* none

## Are there any known issues?

The Conversation logger could use more robust logic, but skipping that for now because ["Slack ghost"](https://github.com/crccheck/messenger-slack-ghost) might be better approach for logging to Slack.

## Did the test coverage decrease?

increases. Some old code is now covered, tests reorganized a bit.
